src/aind_slims_api/behavior_session.py

```diff
@@ -0,0 +1,123 @@
+"""Contains a model for the behavior session content events, a method for
+ fetching it and writing it.
+"""
+
+import logging
+from typing import Any
+from datetime import datetime
+
+from pydantic import Field
+
+from aind_slims_api.core import SlimsBaseModel, SlimsClient, SLIMSTABLES
+
+logger = logging.getLogger()
+
+
+class SlimsBehaviorSessionContentEvent(SlimsBaseModel):
+    """Model for an instance of the Behavior Session ContentEvent"""
+
+    pk: int | None = Field(default=None, alias="cnvn_pk")
+    mouse_name: int | None = Field(
+        default=None, alias="cnvn_fk_content"
+    )  # used as reference to mouse
+    notes: str | None = Field(default=None, alias="cnvn_cf_notes")
+    task_stage: str | None = Field(default=None, alias="cnvn_cf_taskStage")
+    instrument: int | None = Field(default=None, alias="cnvn_cf_fk_instrument")
+    trainers: list[int] = Field(default=[], alias="cnvn_cf_fk_trainer")
+    task: str | None = Field(default=None, alias="cnvn_cf_task")
+    is_curriculum_suggestion: bool | None = Field(
+        default=None, alias="cnvn_cf_stageIsOnCurriculum"
+    )
+    task_schema_version: str | None = Field(
+        default=None, alias="cnvn_cf_taskSchemaVersion"
+    )
+    software_version: str | None = Field(default=None, alias="cnvn_cf_softwareVersion")
+    date: datetime | None = Field(..., alias="cnvn_cf_scheduledDate")
+
+    cnvn_fk_contentEventType: int = 10  # pk of Behavior Session ContentEvent
+
+    _slims_table: SLIMSTABLES = "ContentEvent"
+
+
+SlimsSingletonFetchReturn = SlimsBaseModel | dict[str, Any] | None
+
+
+def _resolve_pk(
+    model: SlimsSingletonFetchReturn,
+    primary_key_name: str = "pk",
+) -> int:
+    """Utility function shared across read/write
+
+    Notes
+    -----
+    - TODO: Change return type of fetch_mouse_content to match pattern in
+     fetch_behavior_session_content_events, or the other way around?
+    - TODO: Move to core to have better centralized control of when references
+     are resolved
+    """
+    if isinstance(model, dict):
+        logger.warning("Extracting primary key from unvalidated dict.")
+        return model[primary_key_name]
+    elif isinstance(model, SlimsBaseModel):
+        return getattr(model, primary_key_name)
+    elif model is None:
+        raise ValueError(f"Cannot resolve primary key from {model}")
+    else:
+        raise ValueError("Unexpected type for model: %s" % type(model))
+
+
+def fetch_behavior_session_content_events(
+    client: SlimsClient,
+    mouse: SlimsSingletonFetchReturn,
```

Comment:
I think it would be better to accept either a SlimsMouseContent or the parameters to fetch it (i.e. barcode)

Reply:
I think that fetching logic for other records shouldn't be embedded in other fetch functions when necessary, it creates unnecessary coupling within our fetching functions. This function is designed to be used like this: 
```python
fetch_behavior_session_content_events(client, fetch_mouse_content(client, labtracks_id), ...)
```
I would prefer to use some record reference abstraction in the future over this pattern but adding record references seems like it goes outside of the bounds of this pr.